planning/behavior_path_planner/src/scene_module/pull_over/shift_pull_over.cpp

```diff
@@ -0,0 +1,359 @@
+  if (pull_over_paths.empty()) {
+    return {};
+  }
+
+  // select valid paths which have enough distance and no lane departure
+  const auto valid_paths = selectValidPaths(
+    pull_over_paths, road_lanes, shoulder_lanes,
+    route_handler->isInGoalRouteSection(road_lanes.back()), goal_pose);
+  if (valid_paths.empty()) {
+    return {};
+  }
+
+  // select safe path
+  for (const auto & path : valid_paths) {
+    if (parameters_.use_occupancy_grid || !occupancy_grid_map_) {
+      const bool check_out_of_range = false;
+      if (occupancy_grid_map_->hasObstacleOnPath(path.shifted_path.path, check_out_of_range)) {
+        continue;
+      }
+    }
+
+    if (parameters_.use_object_recognition) {
+      if (util::checkCollisionBetweenPathFootprintsAndObjects(
+            vehicle_footprint_, path.shifted_path.path, *(planner_data_->dynamic_object),
+            parameters_.object_recognition_collision_check_margin)) {
+        continue;
+      }
+    }
+
+    // found safe path
+    return path;
+  }
+
+  // not found safe path
+  return {};
+}
+
+std::vector<PullOverPath> ShiftPullOver::generatePullOverPaths(
+  const lanelet::ConstLanelets & road_lanes, const lanelet::ConstLanelets & shoulder_lanes,
+  const Pose & goal_pose) const
+{
+  // rename parameter
+  const auto & route_handler = planner_data_->route_handler;
+  const auto & common_parameters = planner_data_->parameters;
+  const Pose & current_pose = planner_data_->self_pose->pose;
+  const double backward_path_length = common_parameters.backward_path_length;
+  const double pull_over_velocity = parameters_.pull_over_velocity;
+  const double after_pull_over_straight_distance = parameters_.after_pull_over_straight_distance;
+  const double margin = parameters_.margin_from_boundary;
+  const double minimum_lateral_jerk = parameters_.minimum_lateral_jerk;
+  const double maximum_lateral_jerk = parameters_.maximum_lateral_jerk;
+  const double deceleration_interval = parameters_.deceleration_interval;
+  const int pull_over_sampling_num = parameters_.pull_over_sampling_num;
+  const double jerk_resolution =
+    std::abs(maximum_lateral_jerk - minimum_lateral_jerk) / pull_over_sampling_num;
+
+  const double distance_to_shoulder_lane_boundary =
+    util::getDistanceToShoulderBoundary(shoulder_lanes, current_pose);
+  const double offset_from_current_pose =
+    distance_to_shoulder_lane_boundary + common_parameters.vehicle_width / 2 + margin;
+
+  // shift end point in shoulder lane
+  PathPointWithLaneId shift_end_point;
+  {
+    const auto arc_position_goal = lanelet::utils::getArcCoordinates(shoulder_lanes, goal_pose);
+    const double s_start = arc_position_goal.length - after_pull_over_straight_distance;
+    const double s_end = s_start + std::numeric_limits<double>::epsilon();
+    const auto path = route_handler->getCenterLinePath(shoulder_lanes, s_start, s_end, true);
+    shift_end_point = path.points.front();
+  }
```

Comment:
seems like `shift_end_point` is not changed anywhere. So why not change the anonymous scope to lambda, and make it `const`.

```suggestion
  const auto shift_end_point = std::invoke([&]() {
    const auto arc_position_goal = lanelet::utils::getArcCoordinates(target_lanelets, goal_pose);
    const double s_start = arc_position_goal.length - after_pull_over_straight_distance;
    const double s_end = s_start + std::numeric_limits<double>::epsilon();
    const auto path = route_handler.getCenterLinePath(target_lanelets, s_start, s_end, true);
    return path.points.front();
  });

```

Reply:
Yes, the value remains the same, it is ust a refactor, To avoid not initializing in clang-tidy, I used lamda, and this value will not be changed so added `const`.

Reply:
I misunderstood, thanks for your suggestion.